src/tla/__rewire_functions_in_apalache.tla

```diff
@@ -0,0 +1,181 @@
+ * Functions. Most importantly, we are adding type annotations. We also
+ * define the Apalache-compatible behavior.
+ *
+ * These definitions are automatically rewired by the Apalache importer.
+ *
+ * Compare with the original definitions in Functions.tla:
+ *
+ * https://github.com/tlaplus/CommunityModules/blob/master/modules/Functions.tla
+ *)
+
+LOCAL INSTANCE Integers
+LOCAL INSTANCE FiniteSets
+LOCAL INSTANCE __apalache_folds 
+
+(**
+ * Restriction of a function to a set (should be a subset of the domain).
+ *
+ * @type: (a -> b, Set(a)) => (a -> b);
+ *)
+Restrict(f, S) == [ x \in S |-> f[x] ]
+
+(**
+ * Range of a function.
+ * Note: The image of a set under function f can be defined as
+ *       Range(Restrict(f,S)).
+ *
+ * @type: (a -> b) => Set(b);
+ *)
+Range(f) == { f[x] : x \in DOMAIN f }
+
+(**
+ * The inverse of a function.
+ * Example:
+ *    LET f == ("a" :> 0 @@ "b" :> 1 @@ "c" :> 2)
+ *    IN Inverse(f, DOMAIN f, Range(f)) =
+ *                                 (0 :> "a" @@ 1 :> "b" @@ 2 :> "c")
+ * Example:
+ *    LET f == ("a" :> 0 @@ "b" :> 1 @@ "c" :> 2)
+ *    IN Inverse(f, DOMAIN f, {1,3}) =
+ *                                 1 :> "b" @@ 3 :> "a")
+ *
+ * @type: (a -> b, Set(a), Set(b)) => (b -> a);
+ *)
+Inverse(f, S, T) == [t \in T |-> CHOOSE s \in S: t \in Range(f) => f[s] = t]
+
+(**
+ * The inverse of a function.
+ *
+ * @type: (a -> b) => (b -> a);
+ *)
+AntiFunction(f) == Inverse(f, DOMAIN f, Range(f))
+
+(**
+ * A function is injective iff it maps each element in its domain to a
+ * distinct element.
+ *
+ * This definition is overridden by TLC in the Java class SequencesExt.
+ * The operator is overridden by the Java method with the same name.
+ *
+ * @type: (a -> b) => Bool;
+ *)
+IsInjective(f) ==
+    \A a, b \in DOMAIN f:
+        f[a] = f[b] => a = b
+
+(**
+ * Set of injections between two sets.
+ *
+ * @type: (Set(a), Set(b)) => Set(a -> b);
+ *)
+Injection(S, T) == { f \in [S -> T]: IsInjective(f) }
+
+(**
+ * This operator is not defined in the community modules.
+ * As a temporary workaround, we define it globally.
+ * It should be declared as LOCAL, once this issue is fixed:
+ * https://github.com/informalsystems/apalache/issues/1495
+ *
+ * @type: (Set(a), Set(b), a -> b) => Bool;
+ *)
+IsSurjective(S, T, f) ==
+    \A t \in T:
+        \E s \in S:
+            f[s] = t
+
+(**
+ * A map is a surjection iff for each element in the range there is some
+ * element in the domain that maps to it.
+ *
+ * @type: (Set(a), Set(b)) => Set(a -> b);
+ *)
+Surjection(S, T) ==
+    { f \in [S -> T]: IsSurjective(S, T, f) }
+
+(**
+ * A map is a bijection iff it is both an injection and a surjection.
+ *
+ * @type: (Set(a), Set(b)) => Set(a -> b);
+ *)
+Bijection(S, T) ==
+    { f \in [S -> T]: IsSurjective(S, T, f) /\ IsInjective(f) }
+
+(**
+ * An injection, surjection, or bijection exists if the corresponding set
+ * is nonempty.
+ *)
+
+(**
```

Comment:
Is this comment meant to refer to the operators before?
If so, could you add it to each of them? Nobody will scroll down to see it.

Reply:
Actually, this comment comes from the community modules, and it is now outdated.